crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -29,121 +29,29 @@ use parity_scale_codec::Encode;
-    pub confirmation_depth_k: u32,
-    /// The size of data in one piece (in bytes).
-    pub record_size: u32,
-    /// Recorded history is encoded and plotted in segments of this size (in bytes).
-    pub recorded_history_segment_size: u32,
-    /// This constant defines the size (in bytes) of one pre-genesis object.
-    pub pre_genesis_object_size: u32,
-    /// This constant defines the number of a pre-genesis objects that will bootstrap the
-    /// history.
-    pub pre_genesis_object_count: u32,
-    /// This constant defines the seed used for deriving pre-genesis objects that will bootstrap
-    /// the history.
-    pub pre_genesis_object_seed: Vec<u8>,
-}
-
-/// Encoded block with mapping of objects that it contains
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct EncodedBlockWithObjectMapping {
-    /// Encoded block
-    #[serde(with = "HexForm")]
-    pub block: Vec<u8>,
-    /// Mapping of objects inside of the block
-    pub object_mapping: BlockObjectMapping,
-}
-
-/// Information about new slot that just arrived
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct RpcNewSlotInfo {
-    /// Slot number
-    pub slot_number: SlotNumber,
-    /// Slot challenge
-    pub challenge: [u8; 8],
-    /// Salt
-    pub salt: [u8; 8],
-    /// Salt for the next eon
-    pub next_salt: Option<[u8; 8]>,
-    /// Acceptable solution range
-    pub solution_range: u64,
-}
-
-/// Information about new slot that just arrived
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct RpcArchivedSegment {
-    /// Root block
-    pub root_block: RootBlock,
-    /// Pieces that correspond to the segment in root block
-    pub pieces: Vec<Vec<u8>>,
-    /// Mappings for objects stored in corresponding pieces.
-    ///
-    /// NOTE: Only first half (data pieces) will have corresponding mapping item in this `Vec`.
-    pub object_mapping: Vec<PieceObjectMapping>,
-}
-
-impl From<ArchivedSegmentNotification> for RpcArchivedSegment {
-    fn from(archived_segment_notification: ArchivedSegmentNotification) -> Self {
-        let ArchivedSegmentNotification {
-            root_block,
-            pieces,
-            object_mapping,
-        } = archived_segment_notification;
-
-        Self {
-            root_block,
-            pieces: pieces.into_iter().map(|piece| piece.to_vec()).collect(),
-            object_mapping,
-        }
-    }
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct RpcSolution {
-    pub public_key: [u8; 32],
-    pub piece_index: u64,
-    pub encoding: Vec<u8>,
-    pub signature: Vec<u8>,
-    pub tag: [u8; 8],
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct ProposedProofOfReplicationResult {
-    pub slot_number: SlotNumber,
-    pub solution: Option<RpcSolution>,
-    pub secret_key: Vec<u8>,
-}
-
 /// Provides rpc methods for interacting with Subspace.
 #[rpc]
-pub trait SubspaceRpcApi {
+pub trait SubspaceApi {
```

Comment:
It was specifically called `SubspaceRpcApi` because we already have `SubspaceApi` is `sp-consensus-subspace` and I believe it was a bit confusing in some context, which is why I decided to disambiguate it.

Reply:
`SubspaceApi` in `sp-consensus-subspace` is a runtime api, we could specify it as `SubspaceRuntimeApi` when we use it. I'd say it's more idiomatic in the substrate context, see all the other RPC impls in substrate. But we could change it back if you insist.

Reply:
Well, I know, but when I read `impl SubspaceApi for Subspace` in RPC layer I have a strange feeling about it. We are not implementing Subspace API for Subspace. What does that even mean?

I can understand if you rename it such that we have `impl RpcApi for RpcHandler` (it would be, potentially, possible to have the same names in other crates upstream for consistency) and we can import them with `Subspace` prefix, but naming a struct `Subspace` definitely doesn't seem right to me.

Following current approach every crate can end up having `Substrate` struct that serves a different purpose. It makes it both harder to read code and harder to use something as simple as text search across the project: just think how many occurrences there will be for `SubspaceRpcHandler` and compare to number of occurrences of `Subspace` in case we rename it as you suggest.

Reply:
Well, that's not a problem for me as each of them are under a specifc contenxt. To some extent, I believe we have so many `Subspace` because we use it as the consensus name and that happens to be the project name :D.

Now I'm getting used to having multiple variant with same name under different contexts. But it's definitely ressonable to keep it more distintive if that helps your dev productivity as we're the developers and we should care about our own dev experience.